pkg/gofr/datasource/file/ftp/fs.go

```diff
@@ -0,0 +1,358 @@
+	return c.ServerConn.Retr(filepath)
+}
+
+func (c *Conn) RetrFrom(filepath string, offset uint64) (ftpResponse, error) {
+	return c.ServerConn.RetrFrom(filepath, offset)
+}
+
+var (
+	errEmptyFilename      = errors.New("filename cannot be empty")
+	errEmptyPath          = errors.New("file/directory path cannot be empty")
+	errEmptyDirectory     = errors.New("directory name cannot be empty")
+	errInvalidArg         = errors.New("invalid filename/directory")
+	transferCompleteError = &textproto.Error{Code: 226, Msg: "Transfer complete."}
+)
+
+// ftpFileSystem represents a file system interface over FTP.
+type fileSystem struct {
+	*file
+	conn    ServerConn
+	config  *Config
+	logger  Logger
+	metrics Metrics
+}
+
+// Config represents the FTP configuration.
+type Config struct {
+	Host      string // FTP server hostname
+	User      string // FTP username
+	Password  string // FTP password
+	Port      int    // FTP port
+	RemoteDir string // Remote directory path. Base Path for all FTP Operations.
+}
+
+// New initializes a new instance of ftpFileSystem with provided configuration.
+func New(config *Config) FileSystemProvider {
+	return &fileSystem{config: config}
+}
+
+// UseLogger sets the logger interface for the FTP file system.
+func (f *fileSystem) UseLogger(logger interface{}) {
+	if l, ok := logger.(Logger); ok {
+		f.logger = l
+	}
+}
+
+// UseMetrics sets the metrics for the MongoDB client which asserts the Metrics interface.
+func (f *fileSystem) UseMetrics(metrics interface{}) {
+	if m, ok := metrics.(Metrics); ok {
+		f.metrics = m
+	}
+}
+
+// Connect establishes a connection to the FTP server and logs in.
+func (f *fileSystem) Connect() {
+	ftpServer := fmt.Sprintf("%v:%v", f.config.Host, f.config.Port)
+
+	defer f.processLog(&FileLog{Operation: "Connect", Location: ftpServer}, time.Now())
+
+	const dialTimeout = 5 * time.Second
+
+	conn, err := ftp.Dial(ftpServer, ftp.DialWithTimeout(dialTimeout))
+	if err != nil {
+		f.logger.Errorf("Connection failed : %v", err)
+		return
+	}
+
+	f.conn = &Conn{conn}
+
+	f.logger.Logf("Connected to FTP Server : %v", ftpServer)
+
+	err = conn.Login(f.config.User, f.config.Password)
+	if err != nil {
+		f.logger.Errorf("Login failed : %v", err)
+		return
+	}
+
+	f.logger.Logf("Login Successful. Current remote location : %q", f.config.RemoteDir)
+}
+
+// Create creates an empty file on the FTP server.
+func (f *fileSystem) Create(name string) (container.File, error) {
+	filePath := path.Join(f.config.RemoteDir, name)
+
+	defer f.processLog(&FileLog{Operation: "Create", Location: filePath}, time.Now())
+
+	if name == "" {
+		f.logger.Errorf("Create_File failed. Provide a valid filename : %v", errEmptyFilename)
+		return nil, errEmptyFilename
+	}
+
+	emptyReader := new(bytes.Buffer)
+
+	err := f.conn.Stor(filePath, emptyReader)
+	if err != nil {
+		f.logger.Errorf("Create_File failed. Error creating file with path %q : %v", filePath, err)
+		return nil, err
+	}
+
+	_, s := path.Split(filePath)
```

Comment:
@puellanivis suggested you using path.Base here

https://github.com/gofr-dev/gofr/pull/871#discussion_r1699879257